src/app/projects/projects.component.ts

```diff
@@ -0,0 +1,99 @@
+import { ChangeDetectorRef, Component, OnInit } from '@angular/core';
+import { ProjectsService } from "./projects.service";
+import { MatBottomSheet, MatDialog } from "@angular/material";
+import { NewProjectComponent } from "./new-project/new-project.component";
+import { Project } from "./project";
+import { DeleteConfirmationDialogComponent } from "../shared/delete-confirmation-dialog/delete-confirmation-dialog.component";
+import { HttpErrorResponse } from "@angular/common/http";
+import { filter, map } from "rxjs/operators";
+import { EditProjectComponent } from "./edit-project/edit-project.component";
+import { GlobalErrorHandlerService } from "../error/global-error-handler.service";
+
+@Component({
+  selector: 'app-projects',
+  templateUrl: './projects.component.html',
+  styleUrls: ['./projects.component.scss']
+})
+export class ProjectsComponent implements OnInit {
+
+  projects: Project[] = [];
+  projectsFiltered: Project[] = [];
+  errorMessage: string = null;
+
+  constructor(private projectService: ProjectsService,
+              private bottomSheet: MatBottomSheet,
+              public dialog: MatDialog,
+              private globalErrorHandlerService: GlobalErrorHandlerService,
+              private changeDetector: ChangeDetectorRef) {
+  }
+
+  ngOnInit() {
+    this.loadProjects();
+  }
+
+  loadProjects() {
+    this.projectService.getProjects()
+      .pipe(map(projects => projects.map<Project>(project => ({
```

Comment:
Is this conversion really necessary? It seems the received structure and the target structure are the same. Can we do without `pipe(map)` stage?

Reply:
done